pkg/manifests/manifests_test.go

```diff
@@ -1194,6 +1194,74 @@ ingress:
 	}
 }
 
+func TestPrometheusQueryLogFileConfig(t *testing.T) {
+	testCases := []struct {
+		name             string
+		queryLogFilePath string
+		expected         string
+		errString        string
+	}{
+		{
+			name:             "basic config",
+			queryLogFilePath: "/tmp/query.log",
+			expected:         "/tmp/query.log",
+			errString:        "",
+		},
+		{
+			name:             "basic config",
+			queryLogFilePath: "/dev/query.log",
+			expected:         "/dev/query.log",
+			errString:        "",
+		},
+		{
+			name:             "basic config",
+			queryLogFilePath: "/prometheus/query.log",
+			expected:         "/prometheus/query.log",
+			errString:        "",
+		},
+		{
+			name:             "basic config",
+			queryLogFilePath: "/query.log",
+			expected:         "",
+			errString:        "attempting to mount query log file on root",
+		},
+	}
+	for _, tt := range testCases {
+		// We need to initialize the volumeClaimTemplate otherwise we will
+		// get a SIGSEGV when creating the EmptyDir volume
```

Comment:
can you elaborate on this issue?

Reply:
Yes, so if I initialize the config with just `c := NewDefaultConfig()` then once the test run I get the error:
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xcf96f5]
```
The error happens on `manifests.go:1682` the line were we see
```
			VolumeSource: v1.VolumeSource{
				EmptyDir: p.Spec.Storage.EmptyDir,
			},
```

Reply:
ack, I've added a comment about emptyDir above that should make it possible to remove the code here.